### ISSUE TITLE
i/o timeout connecting to firehose

### ISSUE THREAD

**Issue by aleliaert** [cloudfoundry-community/firehose-to-syslog#162]
Hi,

We're seeing this error intermittently from our firehose-to-syslog instances running inside Cloud Foundry:

`ERR [2017-11-14 19:29:39.141953315 +0000 UTC] Exception occurred! Message: Error while reading from the firehose Details: read tcp 10.254.0.30:52920->10.150.136.28:443: i/o timeout`

Seems to happen to all instances for a period of several hours, then goes away for days, then comes back for a few hours.  In the meantime, `cf logs` always works to get at app logs, including watching firehose-to-syslog flail trying to connect.  And `cf logs` appears to get its websocket very quickly.

Other info:
* Normally have approx 1200 messages/second
* Using version 3.3.1
* Have tried scaling up, seems to not help
* Currently are at 5 instances of firehose-to-syslog sharing same subscription, 5 traffic controllers, 10 dopplers
* Have F5 in front of gorouters, it's running a newer version of firmware advertised to support websockets
* CPU load for firehose-to-syslog instances, doppler VMs, and trafficontroller VMs is relatively low

Thanks!
Aaron

**Comment by shinji62** [comment]
Hi,

`10.254.0.30:52920` mean you are running f2s as instances  this Ip seems the garden network range ?

To be honest 1200 messages/second is nothing but you syslog server could slow to ingest.
The actual version of f2s could block until message get ingest by your syslog server.


Today I am going to release a new version which support buffering and reconnection to the Firehose.

Thanks.

**Comment by aleliaert** [comment]
Correct, f2s instances are running as an app deployed to Cloud Foundry.

It very well may be that our syslog server is causing backpressure.  We did a lot of testing today and found that the i/o timeout error went away if we wrote to a "fake" syslog server (ncat piping data to /dev/null) or specifying less event types.

We'd be delighted to try a new version.

What are your thoughts on having f2s write event total stats to stdout again?  I recall that an older version did so, which was useful as a sanity check when data was getting lost downstream.  We'd be happy to submit a PR on this.

Thanks for the help, feel free to close this issue as looks like our syslog server.
